source/state_representation/src/space/cartesian/CartesianAcceleration.cpp

```diff
@@ -179,10 +179,19 @@ CartesianAcceleration& CartesianAcceleration::operator+=(const CartesianAccelera
   return (*this);
 }
 
+CartesianAcceleration& CartesianAcceleration::operator+=(const CartesianState& state) {
+  this->CartesianState::operator+=(state);
```

Comment:
suggestion, open to debate: using the base operator in all cases is inefficient because it does extra casting and quaternion manipulations, even though all fields will always be identity. For classes that are not Pose, the operation is trivial and we could do them explicitly here. The only issue is that we lose the "sanity checks" for emptiness in the base method. For this, a protected method `CartesianState::assert_addition_compatibility(a, b)` could be nice.

```suggestion
  this->assert_addition_compatibility(*this, state);
  this->set_acceleration(this->get_acceleration() + state.get_acceleration());
```

But, I admit the downside of repeating the math again here, and it may not be a worthwhile optimization.

Reply:
I would suggest this as another PR. Because it would most probably impact all the other operators, and also we need to discuss it a bit more. My concern is, if we need to change the operation (wrench PR in mind here), I prefer to change it in one place than everywhere where it is repeated. Once we settle on the behavior that we are sure will not change then maybe we should consider it indeed.